zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -376,6 +365,51 @@ impl TransactionRecordsById {
 
         self.invalidate_transactions(txids_to_remove);
     }
+
+    /// Note this method is INCORRECT in the case of a 0-value, 0-fee transaction from the
+    /// Creating Capability.  Such a transaction would violate ZIP317, but could exist in
+    /// the Zcash protocol
+    ///  TODO:   Test and handle 0-value, 0-fee transaction
+    pub(crate) fn transaction_kind(&self, query_record: &TransactionRecord) -> TransactionKind {
+        let mut sapling_spends: Vec<&SaplingNote> =
+            Vec::with_capacity(query_record.spent_sapling_nullifiers.len());
+        query_record
+            .spent_sapling_nullifiers()
+            .iter()
+            .for_each(|nullifier| {
+                if let Some(spend) = self.find_sapling_spend(nullifier) {
+                    sapling_spends.push(spend);
+                }
+            });
+        let mut orchard_spends: Vec<&OrchardNote> =
+            Vec::with_capacity(query_record.spent_orchard_nullifiers.len());
+        query_record
+            .spent_orchard_nullifiers()
+            .iter()
+            .for_each(|nullifier| {
+                if let Some(spend) = self.find_orchard_spend(nullifier) {
+                    orchard_spends.push(spend);
+                }
+            });
+
+        if sapling_spends.is_empty()
+            && orchard_spends.is_empty()
+            && query_record.total_transparent_value_spent == 0
+            && query_record.outgoing_tx_data.is_empty()
```

Comment:
Is there also necessarily an Output created here?

Reply:
sorry i dont understand your question. the logic here says "if no spends have been synced for this tx and there is no outgoing tx data, this is a received tx"

Reply:
we will likely add TEX as a `SendType` variant

Reply:
i think i understand whaty you are saying, that maybe we should explicitly check we received an output? I am happy with this implicit check as there is no way we would be aware of a transaction_record that has no spends or outputs to our capability